lib/puppet/configurer.rb

```diff
@@ -313,19 +322,18 @@ def run_internal(options)
         end
       end
 
-      current_environment = Puppet.lookup(:current_environment)
-      if current_environment.name == @environment.intern
-        local_node_environment = current_environment
-      else
-        local_node_environment = Puppet::Node::Environment.create(@environment,
-                                         current_environment.modulepath,
-                                         current_environment.manifest,
-                                         current_environment.config_version)
+      # This is to maintain compatibility with anyone using this class
+      # aside from agent, apply, device.
+      unless Puppet.lookup(:loaders) { nil }
```

Comment:
@donoghuc @DavidS I noticed ACE [pushes loaders onto the context](https://github.com/puppetlabs/ace/blob/ea30578a8049a371b53ddabffd1575557542c10e/lib/ace/puppet_util.rb#L89), but it omits the `for_agent` argument. Before this PR, the Configurer would always push a loader (with `for_agent = true` which meant ACE could in theory load types, etc from the agent's cache lib directory on the ACE node itself, leading to environment isolation problems. That seems wrong. With this PR, ACE will always use the loaders it pushes, which seems more desirable, but I'm not sure how ACE works to know for sure.